test/OpenTelemetry.Tests/Logs/LogEmitterTests.cs

```diff
@@ -0,0 +1,177 @@
+                    Exception = ex,
+                },
+                new()
+                {
+                    ["key1"] = "value1",
+                    ["key2"] = "value2",
+                });
+
+            Assert.Single(exportedItems);
+
+            var logRecord = exportedItems[0];
+
+            Assert.NotNull(logRecord);
+            Assert.Equal("LogEmitter", logRecord.CategoryName);
+            Assert.Equal("Hello world", logRecord.FormattedMessage);
+            Assert.Equal(LogLevel.Warning, logRecord.LogLevel);
+            Assert.Equal(18, logRecord.EventId.Id);
+            Assert.Equal("CustomEvent", logRecord.EventId.Name);
+            Assert.Equal(ex, logRecord.Exception);
+            Assert.NotEqual(DateTime.MinValue, logRecord.Timestamp);
+
+            Assert.Equal(default, logRecord.TraceId);
+            Assert.Equal(default, logRecord.SpanId);
+            Assert.Equal(ActivityTraceFlags.None, logRecord.TraceFlags);
+            Assert.Null(logRecord.TraceState);
+
+            Assert.NotNull(logRecord.StateValues);
+            Assert.Equal(2, logRecord.StateValues.Count);
+            Assert.Contains(logRecord.StateValues, item => item.Key == "key1" && (string)item.Value == "value1");
+            Assert.Contains(logRecord.StateValues, item => item.Key == "key2" && (string)item.Value == "value2");
+        }
+
+        [Fact]
+        public void LogEmitterFromActivityTest()
+        {
+            var exportedItems = new List<LogRecord>();
+
+            using var provider = new OpenTelemetryLoggerProvider(options =>
+            {
+                options.AddInMemoryExporter(exportedItems);
+            });
+
+            var logEmitter = provider.CreateEmitter();
+
+            using var activity = new Activity("Test");
+
+            activity.Start();
+
+            activity.ActivityTraceFlags = ActivityTraceFlags.Recorded;
+            activity.TraceStateString = "key1=value1";
+
+            logEmitter.Log(new(activity));
+
+            Assert.Single(exportedItems);
+
+            var logRecord = exportedItems[0];
+
+            Assert.NotNull(logRecord);
+
+            Assert.Equal(activity.TraceId, logRecord.TraceId);
+            Assert.Equal(activity.SpanId, logRecord.SpanId);
+            Assert.Equal(activity.ActivityTraceFlags, logRecord.TraceFlags);
+            Assert.Equal(activity.TraceStateString, logRecord.TraceState);
+
+            Assert.Null(logRecord.StateValues);
+        }
+
+        [Fact]
+        public void LogEmitterLocalToUtcTimestampTest()
+        {
+            var exportedItems = new List<LogRecord>();
+
+            using var provider = new OpenTelemetryLoggerProvider(options =>
+            {
+                options.AddInMemoryExporter(exportedItems);
+            });
+
+            var logEmitter = provider.CreateEmitter();
+
+            DateTime timestamp = DateTime.SpecifyKind(
+                new DateTime(2022, 6, 30, 16, 0, 0),
```

Comment:
I'm inclined to suggest using some well-known historical date here, but no great ones are jumping to my mind 😆

Reply:
Are you saying the date that I wrote the test is NOT historically significant?! 🤣